Fix urlencode issues across the library

JacobTaylor-RCC opened the pull request:
Proper fix for the urlencoding issue. Made a utility function for urlencoding. Applied to the database name, and the layout name, which are the two things regularly passed directly into the url. Also applied to container field name, for completeness.

JacobTaylor-RCC committed:
make a prepareURLpart function, wrap layouts in it

Proper fix for the urlencoding issue. Made a utility function for urlencoding. Applied to the database name, and the layout name, which are the two things regularly passed directly into the url. Also applied to container field name, for completeness.

lguilbert commented:
Hi,

See [Pull request #3 ](https://github.com/myFMbutler/myFMApiLibrary-for-PHP/pull/3) to fix that.

steveAllen0112 commented:
> See Pull request #3 to fix that

As you point out in a comment there, @lguilbert, Pull request #3's fix doesn't address everything.

That's why @JacobTaylor-RCC noted exactly that in  [Pull Request #4](https://github.com/myFMbutler/myFMApiLibrary-for-PHP/pull/4).

Then, realizing that the fix would need to be applied everywhere, closed that in favor of this one, which allows for easier future changes to the URL prep logic as necessary.

This PR is the proper one to accept.

JacobTaylor-RCC commented:
Hi @lguilbert optimally you'd merge #3 and then #5 (sorry, #4 was an insufficient solution). If you have improvements here, let me know. I can also add the removal of curl_escape from #3 to this PR so it's prettier if you'd like.

JacobTaylor-RCC committed:
remove curl escape of entire path

if you curl escape the entire path, as this does, the Data API is incapable of processing any request you send. It will encode the slashes in the path to %2d, which chokes the Data API. This, alongside the rawurlencode of the relevant url parts (via prepareURLpart utility function), should resolve that.
